File: Examples/VisualBasic/ProgrammersGuide/Working-with-Masters/GetMasterbyName.vb
Imports VisualBasic
Imports Aspose.Diagram
Imports System

Public Class GetMasterbyName
    Public Shared Sub Run()
        'ExStart:GetMasterbyName
        ' The path to the documents directory.
        Dim dataDir As String = RunExamples.GetDataDir_Master()

        ' Call the diagram constructor to load diagram from a VDX file
        Dim diagram As New Diagram(dataDir & Convert.ToString("Basic Shapes.vss"))

        ' Set master name
        Dim masterName As String = "Circle"
        ' Get master object by name
        Dim master As Master = diagram.Masters.GetMasterByName(masterName)

        Console.WriteLine("Master ID : " & master.ID)
        Console.WriteLine("Master Name : " & master.Name)
        Console.WriteLine("Master Name : " & master.UniqueID.ToString())
        'ExEnd:GetMasterbyName
    End Sub
End Class

File: Examples/CSharp/Working-with-Masters/GetMasterbyName.cs
using Aspose.Diagram;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Aspose.Diagram.Examples.CSharp.Working_with_Masters
{
    public class GetMasterbyName
    {
        public static void Run()
        {
            // ExStart:GetMasterbyName
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_Master();

            // Call the diagram constructor to load diagram from a VDX file
            Diagram diagram = new Diagram(dataDir + "Basic Shapes.vss");

            // Set master name
            string masterName = "Circle";
            // Get master object by name
            Master master = diagram.Masters.GetMasterByName(masterName);

            Console.WriteLine("Master ID : " + master.ID);
            Console.WriteLine("Master Name : " + master.Name);
            Console.WriteLine("Master Name : " + master.UniqueID);
            // ExEnd:GetMasterbyName
        }
    }
}
